code/game/objects/effects/decals/Cleanable/humans.dm

```diff
@@ -230,6 +231,18 @@
 
 		if (step_to(src, get_step(src, direction), 0))
 			break
+//This proc prevents blood on openspace tiles, by causing them to fall down until they hit the ground
+/obj/effect/decal/cleanable/blood/proc/fall_to_floor()
+	if (isopenturf(loc))
+		anchored = FALSE //Anchored things can't fall
+		while (isopenturf(loc))

```

Comment:
This doesn't seem like a smart idea. If you invoke this in a setting when the falling ticker isn't work, you'll get an infinite loop. It's also logically redundant to do this in a loop. It should be enough to un-anchor it and to make it fall once. You can then override the proc which handles hitting the ground to resent `anchored = 1`.

The alternative is, if you want this to be instant, which it perhaps should be, is to bypass the falling API and to just keep getting the turf below it until you hit one that isn't openturf, or you run out of space (at which point, qdel).